src/serde_value_ext.rs

```diff
@@ -0,0 +1,154 @@
+use anyhow::Result;
+use async_graphql::Value as GraphQLValue;
+use indexmap::IndexMap;
+use serde_json::Value;
+
+use crate::mustache::Mustache;
+use crate::path::PathString;
+
+pub trait ValueExt {
+    fn render_value(&self, ctx: &impl PathString) -> Result<GraphQLValue>;
+}
+
+fn eval_types(value: &Value) -> Result<GraphQLValue> {
+    match value {
+        Value::Array(arr) => {
+            let mut out = Vec::new();
+            for v in arr {
+                let value = eval_types(v)?;
+                out.push(value);
+            }
+            Ok(async_graphql::Value::List(out))
+        }
+        Value::String(s) => {
+            let out = serde_json::from_str::<Value>(s.as_str());
+            match out {
+                Ok(v) => async_graphql::Value::from_json(v).map_err(|e| anyhow::anyhow!(e)),
+                Err(_) => Ok(async_graphql::Value::String(s.to_owned())),
+            }
+        }
+        _ => async_graphql::Value::from_json(value.clone()).map_err(|e| anyhow::anyhow!(e)),
+    }
+}
+
+fn string_to_value(s: String) -> Value {
+    let out = serde_json::from_str::<Value>(&s);
+    match out {
+        Ok(v) => v,
+        Err(_) => Value::String(s),
+    }
+}
+
+impl ValueExt for Value {
+    // Optimize render_value to avoid unnecessary conversions
+    fn render_value(&self, ctx: &impl PathString) -> Result<GraphQLValue> {
+        match self {
+            Value::Object(obj) => {
+                let mut out = IndexMap::new();
+                for (k, v) in obj {
+                    let value = Mustache::parse(v.to_string().as_str())?;
```

Comment:
I feel like here we don't need to do any parsing and just need to call this function recursively on child fields until we face the string actually.

Because here in case of nested object we will convert this to json, parse it as mustache, render and parse the result json back again to value. It should be if string parse it as mustache, render parse result as json I think

Reply:
moved parsing to compile time